Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,326 @@
+    /// <param name="body">The body content.</param>
+    /// <param name="isBase64Encoded">Whether the body is Base64 encoded.</param>
+    private static void SetResponseBody(HttpResponse response, string? body, bool isBase64Encoded, ApiGatewayEmulatorMode apiGatewayEmulator)
+    {
+        if (!string.IsNullOrEmpty(body))
+        {
+            byte[] bodyBytes;
+            if (isBase64Encoded && ApiGatewayEmulatorMode.Rest != apiGatewayEmulator)
+            {
+                bodyBytes = Convert.FromBase64String(body);
+            }
+            else
+            {
+                bodyBytes = Encoding.UTF8.GetBytes(body);
+            }
+
+            response.Body = new MemoryStream(bodyBytes);
+            response.ContentLength = bodyBytes.Length;
+        }
+    }
+
+    /// <summary>
+    /// Sets the content type and status code for API Gateway v1 responses.
+    /// </summary>
+    /// <param name="response">The HttpResponse to set the content type and status code on.</param>
+    /// <param name="headers">The single-value headers.</param>
+    /// <param name="multiValueHeaders">The multi-value headers.</param>
+    /// <param name="statusCode">The status code to set.</param>
+    private static void SetContentTypeAndStatusCode(HttpResponse response, IDictionary<string, string>? headers, IDictionary<string, IList<string>>? multiValueHeaders, int statusCode, ApiGatewayEmulatorMode emulatorMode)
+    {
+        string? contentType = null;
+
+        if (headers != null && headers.TryGetValue("Content-Type", out var headerContentType))
+        {
+            contentType = headerContentType;
+        }
+        else if (multiValueHeaders != null && multiValueHeaders.TryGetValue("Content-Type", out var multiValueContentType))
+        {
+            contentType = multiValueContentType[0];
+        }
+
+        if (contentType != null)
+        {
+            response.ContentType = contentType;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.HttpV1)
+            {
+                response.ContentType = "text/plain; charset=utf-8";
+            } 
+            else if (emulatorMode == ApiGatewayEmulatorMode.Rest)
+            {
+                response.ContentType = "application/json";
+            }
+        }
+
+        if (statusCode != 0)
+        {
+            response.StatusCode = statusCode;
+        }
+        else
+        {
+            if (emulatorMode == ApiGatewayEmulatorMode.Rest) // rest api text for this message/error code is slightly different
+            {
+                response.StatusCode = 502;
+                response.ContentType = "application/json";
+                var errorBytes = Encoding.UTF8.GetBytes("{\"message\": \"Internal server error\"}");
+                response.Body = new MemoryStream(errorBytes);
+                response.ContentLength = errorBytes.Length;
+            } else
```

Comment:
nit: Please put the `else` on a separate line.

Reply:
fixed in latest revision